src/kv/kv_types.h

```diff
@@ -50,8 +50,53 @@ namespace kv
   DECLARE_JSON_TYPE(TxID);
   DECLARE_JSON_REQUIRED_FIELDS(TxID, term, version)
 
-  using BatchVector = std::vector<
-    std::tuple<kv::Version, std::shared_ptr<std::vector<uint8_t>>, bool>>;
+  // SeqNo indexes transactions processed by the consensus protocol providing
+  // ordering
+  using SeqNo = int64_t;
+
+  struct Configuration
+  {
+    struct NodeInfo
+    {
+      std::string hostname;
+      std::string port;
```

Comment:
should this not be a uint16_t?

Reply:
It's resolved further down, this is still raw from the KV at this point. Technically, Unix allows named ports (/etc/services), and I expect libuv probably handles them (or rather, does nothing to prevent them from working). They're rather out of fashion these days.